### PR TITLE
Add commit hash/version in footer

### PR DESCRIPTION
**Pull request by fstachura**

### CODE REVIEW

File: elixir/web_utils.py [hunk 1]
```diff
@@ -1,8 +1,39 @@
...
+    try:
+        # try to get Elixir version from git
+        result, return_code = run_cmd('git',
+            '-C', ELIXIR_DIR,
+            '-c', f'safe.directory={ os.path.normpath(ELIXIR_DIR) }',
+            'rev-parse', '--short', 'HEAD'
+        )
+
+        if return_code == 0:
+            return result.decode('utf-8')
```
[the remark this answers is not in the document]
Comment (fstachura) [reply elixir/web_utils.py:24]: run_cmd logs errors

File: elixir/web_utils.py [hunk 2]
```diff
@@ -1,8 +1,39 @@
...
+
+    except Exception:
+        logging.exception("failed to get elixir commit hash")
+
+    return 'v2.2'
```
[the remark this answers is not in the document]
Comment (fstachura) [reply elixir/web_utils.py:29]: Changed in next iteration

File: elixir/web_utils.py [hunk 3]
```diff
@@ -1,8 +1,39 @@
...
+
+def get_elixir_repo_link(version):
+    if re.match('^[0-9a-f]{5,12}$', version) or version.startswith('v'):
+        return f'https://github.com/bootlin/elixir/tree/{ version }'
+    else:
+        return f'https://github.com/bootlin/elixir/'
```
[the remark this answers is not in the document]
Comment (fstachura) [reply elixir/web_utils.py:35]: I think this is useful, and when I see a commit hash, I would expect to be redirected to a particular commit. The link to a particular version can be separate if you think this helps.

[the remark this answers is not in the document]
Comment (fstachura) [reply elixir/web_utils.py:35]: The only problem is that the linked version may not exist in the upstream repo if someone hosts their own. But maybe this is actually good, it signifies that it's a custom instance

File: elixir/web_utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tleb) [reply elixir/web_utils.py:26]: No: `run_cmd()` logs only when something is emitted to stderr. Something can fail (`return_code != 0`) but not emit anything on stderr.

I think `get_elixir_version_string()` should be more like (this is a proof-of-concept, not tested):

```python
def get_elixir_version_string():
    version = os.environ.get('ELIXIR_VERSION')
    if version is not None and len(version) != 0:
        return version

    try:
        # try to get Elixir version from git
        result, return_code = run_cmd('git',
            '-C', ELIXIR_DIR,
            '-c', f'safe.directory={ os.path.normpath(ELIXIR_DIR) }',
            'rev-parse', '--short', 'HEAD'
        )

        if return_code != 0:
            raise ValueError(f"git rev-parse returned {return_code}")

        return result.decode('utf-8')
    except Exception as e:
        logging.exception(f"failed to get elixir commit hash: {e}")

    return ''
```

**Comment by tleb** [comment]
Merged as cb28f73dbc06b08dea76d2984d63ae5f998a92cd + 305395e8e9aa154f06c5432f2dce8d3251a2af15 + 6b47af27a758f00037c7dd1a74f84fe0954aee35 & deployed, thanks!

**Pull request closed without merging**